SleepEngine/Engine/Jobs/JobThread.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include <Engine/Jobs/ThreadSafeQueue.h>
+#include <Engine/Jobs/Job.h>
+#include <Engine/Jobs/Event.h>
+
+BEGIN_NAMESPACE_SLEEP
+
+class JobThread
+{
+public:
+    using JobQueue = ThreadSafeQueue<std::unique_ptr<Job>>;
+
+    // must pass Event because it cannot be part of thread -
+    // thread must be movable to support addition to vector, and Event is neithre movable nor copyable,
+    // so must be stored in a static default constructed vector somewhere else
+    JobThread(JobQueue& jobQueue, Event& workAvailable, std::atomic<bool> const& shutdownRequested);
+
+    REF_GETTER(getWorkAvailable, m_workAvailable);
+
+    void start();
+    // called from the worker thread, before polling
+    virtual void init() {}
+
+    void join() { m_thread.join(); }
+    bool joinable() const { return m_thread.joinable(); }
+
+private:
+    Event& m_workAvailable;
+    std::atomic<bool> const& m_shutdownRequested;

```

Comment:
Are you sure it is a good idea to store a reference on atomic bool?
Looks very bad

Reply:
Don't think there is smth wrong with this, I need to modify it in one class and access in another.
It looks strange because we're accustomed to singletons, but this is actually the more safe and explicit way to do it.